File: results/transcoder/translations/py/COUNT_SUBARRAYS_EQUAL_NUMBER_1S_0S.cpp
int countSubarrWithEqualZeroAndOne ( int arr [ ] , int n ) {
  map < int , int > um ;
  int currSum = 0 ;
  for ( int i = 0 ;
  i < n ;
  i ++ ) {
    currSum += ( - 1 == ( arr [ i ] == 0 ) ) ? arr [ i ] : arr [ i ] ;
    if ( um . count ( currSum ) ) um [ currSum ] ++ ;
    else um [ currSum ] = 1 ;
  }
  int count = 0 ;
  for ( auto itr : um ) {
    if ( um [ itr ] > 1 ) count += ( ( um [ itr ] * ( int ) ( um [ itr ] - 1 ) ) / 2 ) ;
  }
  if ( um . count ( 0 ) ) count += um [ 0 ] ;
  return ( int ) count ;
}

File: cfgs/java_cfgs_foo/COUNT_SUBARRAYS_EQUAL_NUMBER_1S_0S.java
Nodes
1;special;;;
2;statement;Map<Integer, Integer> um = new HashMap<>();14;
3;statement;int curr_sum = 0;15;
4;statement;int i = 0;16;
5;conditional;foo_1 < foo_2;17;
6;statement;curr_sum += (foo_1[foo_2] == 0) ? -1 : foo_3[foo_4];19;
7;statement;fun_1(foo_2, fun_2(foo_4) == null ? 1 : fun_3(foo_6) + 1);20;
8;statement;i++;18;
9;statement;int count = 0;22;
10;statement;Map.Entry<Integer, Integer> itr;23;
11;conditional;fun_1();23;
12;conditional;fun_1() > 1;24;
13;statement;count += (fun_1() * (fun_2() - 1)) / 2;24;
14;conditional;fun_1(0);26;
15;statement;count += fun_1(0);26;
16;statement;return foo_1;27;
17;exit;;;
Edges
0;2;
1;3;
2;4;
3;5;
4;6;9;
5;7;
6;8;
7;5;
8;10;
9;11;
10;12;14;
11;13;11;
12;11;
13;15;16;
14;16;
15;17;
16;